File: Scriptable/py-scripts/mmios_sign.py
# /home/Takagivegeta/py-scripts/mmios_sign.py
import requests
import re
import logging

class MMIOS:
    def __init__(self):
        self.base_url = "https://www.mmios.net"
        self.ajax_url = f"{self.base_url}/wp-admin/admin-ajax.php"
        self.session = requests.Session()
        self.ua = "Mozilla/5.0 (iPhone; CPU iPhone OS 18_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.0 Mobile/15E148 Safari/604.1"
        self.session.headers.update({"User-Agent": self.ua})

    def run(self):
        try:
            # 1. 登录
            login_data = {
                "action": "user_login",
                "username": "",
                "password": "",
                "rememberme": "1"
            }
            login_res = self.session.post(self.ajax_url, data=login_data, headers={"X-Requested-With": "XMLHttpRequest"}).json()
            
            if login_res.get("status") != "1":
                return f"<b>mmios.net</b>\n状态：登录失败\n详情：{login_res.get('msg')}"

            # 2. 获取签到 Nonce
            # 原脚本在 /tuiguang 页面获取 nonce
            page_r = self.session.get(f"{self.base_url}/tuiguang")
            nonce_match = re.search(r'data-nonce\s*=\s*["\']([a-f0-9]{10})["\']', page_r.text, re.I)
            if not nonce_match:
                return "<b>mmios.net</b>\n状态：失败 (未找到Nonce)"
            nonce = nonce_match.group(1)

            # 3. 执行签到
            sign_data = {
                "action": "user_qiandao",
                "nonce": nonce
            }
            sign_res = self.session.post(self.ajax_url, data=sign_data, headers={
                "X-Requested-With": "XMLHttpRequest",
                "Referer": f"{self.base_url}/tuiguang"
            }).json()

            msg = sign_res.get("msg", "未知结果")
            if sign_res.get("status") == "1":
                return f"<b>mmios.net</b>\n状态：签到成功 🎉\n详情：{msg}"
            else:
                return f"<b>mmios.net</b>\n状态：提示\n详情：{msg}"

        except Exception as e:
            return f"<b>mmios.net</b>\n状态：脚本异常\n详情：{str(e)}"

def main_task():
    client = MMIOS()
    return client.run()

if __name__ == "__main__":
    print(main_task())

File: Scriptable/mmios_sign.js
// ==Scriptable==
// mmios.net 自动签到
// @description  自动登录并每日签到，获取积分
// @author       Takagivegeta
// ==/Scriptable===

const USERNAME = "";
const PASSWORD = "";

const LOGIN_URL = "https://www.mmios.net/wp-admin/admin-ajax.php";
const TUIGUANG_URL = "https://www.mmios.net/tuiguang";
const QIAND_AO_URL = "https://www.mmios.net/wp-admin/admin-ajax.php";

async function main() {
  let request = new Request(LOGIN_URL);
  request.method = "POST";
  request.headers = {
    "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
    "X-Requested-With": "XMLHttpRequest",
    "User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 18_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.0 Mobile/15E148 Safari/604.1",
    "Origin": "https://www.mmios.net",
    "Referer": "https://www.mmios.net/user"
  };
  request.body = `action=user_login&username=${encodeURIComponent(USERNAME)}&password=${encodeURIComponent(PASSWORD)}&rememberme=1`;
  let loginResponse = await request.loadJSON();
  if (loginResponse.status !== "1") {
    await notify("mmios.net-登录失败", loginResponse.msg);
    return;
  }
  await notify("mmios.net-登录成功", "开始获取签到 nonce");
  let pageRequest = new Request(TUIGUANG_URL);
  pageRequest.headers = {
    "User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 18_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.0 Mobile/15E148 Safari/604.1"
  };
  let html = await pageRequest.loadString();
  let nonceMatch = html.match(/data-nonce\s*=\s*["']([a-f0-9]{10})["']/i);
  if (!nonceMatch || !nonceMatch[1]) {
    await notify("mmios.net-获取 nonce 失败", "页面中未找到 data-nonce");
    return;
  }
  let nonce = nonceMatch[1];
  await notify("mmios.net-获取 nonce 成功", nonce);
  let qianRequest = new Request(QIAND_AO_URL);
  qianRequest.method = "POST";
  qianRequest.headers = {
    "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
    "X-Requested-With": "XMLHttpRequest",
    "User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 18_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.0 Mobile/15E148 Safari/604.1",
    "Origin": "https://www.mmios.net",
    "Referer": TUIGUANG_URL
  };
  qianRequest.body = `action=user_qiandao&nonce=${nonce}`;
  let qianResponse = await qianRequest.loadJSON();
  if (qianResponse.status === "1") await notify("mmios.net-签到成功 🎉", qianResponse.msg);
  else await notify("mmios.net-签到失败", qianResponse.msg);
}

async function notify(title, body) {
  let notification = new Notification();
  notification.title = title;
  notification.body = body;
  await notification.schedule();
}

await main();
